cpp/src/arrow/acero/exec_plan.cc

```diff
@@ -951,7 +951,6 @@ struct BatchConverter {
     if (exec_plan->finished().is_finished()) {
       return;
     }
-    exec_plan->StopProducing();
```

Comment:
Why removing this?

Reply:
It's moved, not removed. See [this post](https://github.com/apache/arrow/pull/35953/#issuecomment-1594187748).

Reply:
I am not sure about this. It's not clear to me what's the intent of calling StopProducing here in the first place. And `reader->Close()` isn't called in all cases so I think it is safer to leave this unchanged.

Reply:
Reinstating this invocation requires a guarantee that `ExecPlan::StopProducing()` is idempotent, otherwise if `PlanReader::Close()` occurs then the invocation here which must occur as well may misbehave. I don't see this in guarantee in the documentation. @westonpace, WDYT?

Reply:
@rtpsw Can you try to figure out if ExecPlan::StopProducing() is idempotent

Reply:
Looking at `StopProducing`, it has an atomic-bool guarding against double-execution, I'll reinstate.